modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy

```diff
@@ -252,17 +262,20 @@ class CondaCache {
         log.info "Creating env using ${binaryName}: $condaEnv [cache $prefixPath]"
 
         final opts = createOptions ? "$createOptions " : ''
+        // micromamba does not and might never support the mkdir flag, since the mkdir behaviour is the default
+        final mkdir = binaryName == "micromamba" ? '' : '--mkdir'
+
         def cmd
         if( isYamlFilePath(condaEnv) ) {
             cmd = "${binaryName} env create --prefix ${Escape.path(prefixPath)} --file ${Escape.path(makeAbsolute(condaEnv))}"
         }
         else if( isTextFilePath(condaEnv) ) {
 
-            cmd = "${binaryName} create $opts--mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} --file ${Escape.path(makeAbsolute(condaEnv))}"
+            cmd = "${binaryName} create $opts $mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} --file ${Escape.path(makeAbsolute(condaEnv))}"
         }
 
         else {
-            cmd = "${binaryName} create $opts--mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} $condaEnv"
+            cmd = "${binaryName} create $opts $mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} $condaEnv"
```

Comment:
This is not a mistake. the `opts` is expected to have a trailing space.

Reply:
Yes thanks, I just saw it when I re-ran the tests. It has double spaces now.